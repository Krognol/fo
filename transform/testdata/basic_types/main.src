package main

type A[T] T

func main() {
	var a0 A[string] = "foo"
	var _ string = a0
	var a1 A[int] = 42
	var _ int = a1
	a2 := A[string]("foo")
	print(a2)
	var a3 A[[]int] = []int{1, 2, 3}
	var _ int = a3[0]
	var a4 A[map[string]int] = map[string]int{
		"bar": 42,
	}
	var _ int = a4["bar"]
}
