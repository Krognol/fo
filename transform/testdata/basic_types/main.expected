package main

type A interface{}

func main() {
	var a0 A = "foo"
	var _ string = a0.(string)
	var a1 A = 42
	var _ int = a1.(int)
	a2 := A("foo")
	print(a2.(string))
	var a3 A = []int{1, 2, 3}
	var _ int = a3.([]int)[0]
	var a4 A = map[string]int{
		"bar": 42,
	}
	var _ int = a4.(map[string]int)["bar"]
}
